manifests/forwarder.pp

```diff
@@ -172,41 +172,41 @@
-  Optional[String[1]] $package_source             = undef,
-  Variant[Hash, Array[String[1]]]$install_options = $splunk::params::forwarder_install_options, # Not strongly typed because can be Hash or Array
-  String[1] $splunk_user                          = $splunk::params::splunk_user,
-  Stdlib::Absolutepath $forwarder_homedir         = $splunk::params::forwarder_homedir,
-  Stdlib::Absolutepath $forwarder_confdir         = $splunk::params::forwarder_confdir,
-  String[1] $service_name                         = $splunk::params::forwarder_service,
-  Stdlib::Absolutepath $service_file              = $splunk::params::forwarder_service_file,
-  Boolean $boot_start                             = $splunk::params::boot_start,
-  Boolean $use_default_config                     = true,
-  Stdlib::IP::Address $splunkd_listen             = '127.0.0.1',
-  Stdlib::Port $splunkd_port                      = $splunk::params::splunkd_port,
-  Stdlib::Port $logging_port                      = $splunk::params::logging_port,
-  Boolean $purge_deploymentclient                 = false,
-  Boolean $purge_outputs                          = false,
-  Boolean $purge_inputs                           = false,
-  Boolean $purge_props                            = false,
-  Boolean $purge_transforms                       = false,
-  Boolean $purge_web                              = false,
-  Hash $forwarder_output                          = $splunk::params::forwarder_output,
-  Hash $forwarder_input                           = $splunk::params::forwarder_input,
-  Boolean $manage_password                        = $splunk::params::manage_password,
-  Stdlib::Absolutepath $password_config_file      = $splunk::params::forwarder_password_config_file,
-  String[1] $password_content                     = $splunk::params::password_content,
-  Stdlib::Absolutepath $secret_file               = $splunk::params::forwarder_secret_file,
-  String[1] $secret                               = $splunk::params::secret,
-  Hash $addons                                    = {},
+  String[1] $server                          = $splunk::params::server,
+  String[1] $version                         = $splunk::params::version,
+  String[1] $package_name                    = $splunk::params::forwarder_package_name,
+  String[1] $package_ensure                  = $splunk::params::forwarder_package_ensure,
+  String[1] $staging_dir                     = $splunk::params::staging_dir,
+  String[1] $path_delimiter                  = $splunk::params::path_delimiter,
+  String[1] $forwarder_package_src           = $splunk::params::forwarder_package_src,
+  Optional[String[1]] $package_provider      = $splunk::params::package_provider,
+  Boolean $manage_package_source             = true,
+  Optional[String[1]] $package_source        = undef,
+  Variant[Tuple[Hash,String,1,default], Array[String[1]]]$install_options = $splunk::params::forwarder_install_options,
```

Comment:
ooh.  Since this is turning into quite a handful, it might make sense to create a custom type alias, (eg `Splunk::ForwarderInstallOptions`) ?  These can be tested in isolation. https://github.com/rodjek/rspec-puppet#type-alias-matchers

Reply:
Quite possibly, but that's above my skill level unfortunately.

Reply:
@kreeuwijk No worries.  We can always fix that up later.  I'm going to mark this WIP so the PR doesn't accidentally get merged before you're happy you're finished.  Just remove the WIP from the PR title when you're done.

Reply:
@alexjfisher ready for merging.